az-core/src/main/java/azkaban/utils/Utils.java

```diff
@@ -198,6 +200,14 @@ public static void unzip(final ZipFile source, final File dest) throws IOExcepti
     final Enumeration<?> entries = source.entries();
     while (entries.hasMoreElements()) {
       final ZipEntry entry = (ZipEntry) entries.nextElement();
+      final Path fileDestinationPath = Paths.get(dest.getAbsolutePath(),
+          entry.getName()).normalize();
```

Comment:
How thorough is the `normalize()` method?

Looking at the documentation, it looks like it handles `.` and `..`, but does it handle `~`? How does it handle symlinks? (i.e. what if there's a symlink in the zip file that points to root, then the file leverages that symlink?)

Reply:
Good question. As discussed offline, "normalize()" does not handle "~". If there is "~" in the file name, it will just directly interpret as the folder name "~". 
E.g. if dest path is "/user/foo" and entry name is "~/bar.txt", after calling normalize(), fileDestinationPath would be "/user/foo/~/bar.txt". It will still create the unzipped file in the current directory, which should not be a problem. 
"normalize()" does not resolve symlinks. Our current unzipping method only handles files and directories. So even if there are evil symlinks, we will not be trapped.